Newspack/Newspack/Stores/AssetStore.swift

```diff
@@ -0,0 +1,58 @@
+import Foundation
+import WordPressFlux
+
+/// Responsible for managing folder related things.
+///
+class AssetStore: Store {
+
+    private(set) var currentSiteID: UUID?
+
+    private let folderManager: FolderManager
+
+    private var selectedFolder: URL?
+
+    init(dispatcher: ActionDispatcher = .global, siteID: UUID? = nil) {
+        currentSiteID = siteID
+
+        folderManager = SessionManager.shared.folderManager
+
+        if let _ = siteID, let folder = StoreContainer.shared.folderStore.listFolders().first {
+            selectedFolder = folder
+        }
+
+        super.init(dispatcher: dispatcher)
+    }
+
+    /// Action handler
+    ///
+    override func onDispatch(_ action: Action) {
+        if let action = action as? AssetAction {
+            switch action {
+            case .selectFolder(let folder) :
+                selectFolder(folder: folder)
+            }
+        }
+    }
+}
+
+extension AssetStore {
+
+    func getSelectedFolder() -> URL? {
```

Comment:
Maaaybe patch the private property, and drop this getter?

```
private(set) var selectedFolder: URL?
```

Reply:
Yeah. Good call. I'd started in a different direction than where I ended up and the getter method is a left over from that.